src/modules/projects/components/tables/ProjectExternalSubmissionsTable.tsx

```diff
@@ -0,0 +1,198 @@
+import GenericTableWithData from '@/modules/dataFetching/components/GenericTableWithData';
+import DynamicView from '@/modules/form/components/viewable/DynamicView';
+import { useStaticFormDialog } from '@/modules/form/hooks/useStaticFormDialog';
+import { FormValues } from '@/modules/form/types';
+import { getItemMap } from '@/modules/form/util/formUtil';
+import {
+  customDataElementValueForKey,
+  getCustomDataElementColumns,
+  parseAndFormatDateTime,
+} from '@/modules/hmis/hmisUtil';
+import { cache } from '@/providers/apolloClient';
+import {
+  DeleteExternalFormSubmissionDocument,
+  DeleteExternalFormSubmissionMutation,
+  DeleteExternalFormSubmissionMutationVariables,
+  ExternalFormSubmissionFieldsFragment,
+  ExternalFormSubmissionFilterOptions,
+  ExternalFormSubmissionInput,
+  ExternalFormSubmissionStatus,
+  GetProjectExternalFormSubmissionsDocument,
+  GetProjectExternalFormSubmissionsQuery,
+  GetProjectExternalFormSubmissionsQueryVariables,
+  StaticFormRole,
+  UpdateExternalFormSubmissionDocument,
+  UpdateExternalFormSubmissionMutation,
+  UpdateExternalFormSubmissionMutationVariables,
+} from '@/types/gqlTypes';
+
+export type ExternalFormSubmissionFields = NonNullable<
+  GetProjectExternalFormSubmissionsQuery['project']
+>['externalFormSubmissions']['nodes'][number];
+
+const ProjectExternalSubmissionsTable = ({
+  projectId,
+  formDefinitionIdentifier,
+}: {
+  projectId: string;
+  formDefinitionIdentifier: string;
+}) => {
+  const getColumnDefs = useCallback((rows: ExternalFormSubmissionFields[]) => {
+    const customColumns = getCustomDataElementColumns(rows);
+    return [
+      {
+        header: 'Status',
+        linkTreatment: false,
+        render: (s: ExternalFormSubmissionFieldsFragment) => {
+          const isNew = s.status === ExternalFormSubmissionStatus.New;
+          return (
+            <Chip
+              label={capitalize(s.status)}
+              size='small'
+              color={isNew ? 'primary' : 'default'}
+              variant={isNew ? 'filled' : 'outlined'}
+              sx={isNew ? {} : { color: theme.palette.text.secondary }}
+            />
+          );
+        },
+      },
+      {
+        header: 'Date Submitted',
+        linkTreatment: false,
+        render: (s: ExternalFormSubmissionFieldsFragment) =>
+          parseAndFormatDateTime(s.submittedAt),
+      },
+      ...customColumns,
+    ];
+  }, []);
+
+  const [selected, setSelected] =
+    useState<ExternalFormSubmissionFieldsFragment | null>(null);
+
+  const submissionValues = useMemo(() => {
+    if (!selected) return {};
+    const itemMap = getItemMap(selected.definition.definition, true);
+    const submissionValues: FormValues = {};
+    Object.keys(itemMap).forEach((key) => {
+      const pickListOptions = itemMap[key].pickListOptions;
+      const customFieldKey = itemMap[key].mapping?.customFieldKey;
+      if (!customFieldKey) return;
+
+      const value = customDataElementValueForKey(
```

Comment:
This could be a single value or an array or values, depending on whether the custom field `repeats` or not. (That would be like a multi-select dropdown).

To accommodate that (untested):
```
# if item has a picklist, convert value to PickListOption(s) so we can display the readable label
if (item.pickListOptions) {
  if (Array.isArray(value)) {
     submissionValues[key] = value.map(v => getOptionValue(v, item);
  } else {
     submissionValues[key] = getOptionValue(value, item);
  }
} else {
   submissionValues[key] = value;
}
```

Reply:
I tested this out by locally setting one of the dropdowns to have `repeats` as true and then adding an extra CDE value for it in the db. It looked good to me!